packit/config/package_config.py

```diff
@@ -253,6 +273,41 @@ def __eq__(self, other: object):
             and self.upstream_tag_template == other.upstream_tag_template
         )
 
+    def __setitem__(self, key, value):
+        setattr(self, key, value)
+
+    def __getitem__(self, key, default=None):
+        if key == "dist_git_url":
+            raise KeyError()
+        return self.__dict__.get(key, default)
+
+    def __len__(self):
+        return len(self.__dict__)
+
+    def __iter__(self):
+        package_dict = self.__dict__
+        del package_dict["_downstream_project_url"]
+        del package_dict["dist_git_clone_path"]
```

Comment:
with this you are removing the attributes for good, since self.__dict__ is mutable; also why they need to be removed?

The best practice here is to copy self.__dict__

```python
package_dict = deepcopy(self.__dict__)